service/history/historyEngine.go

```diff
@@ -2254,20 +2255,93 @@ type updateWorkflowAction struct {
 	createDecision bool
 }
 
-func (e *historyEngineImpl) updateWorkflowExecutionWithAction(
+type workflowLockedContext struct {
+	context workflowExecutionContext
+	release releaseWorkflowExecutionFunc
+	err     error
+}
+
```

Comment:
can we put this is a dedicated file? preferably workflowExecutionUtil

Reply:
ok